invenio_app_ils/records/serializers/loan.py

```diff
@@ -38,8 +39,29 @@ def transform_search_hit(
             pid, record_hit, links_factory=links_factory, **kwargs
         )
         self.is_overdue(hit["metadata"])
+        self.document_cover_urls(hit["metadata"])
         return hit
 
+    def document_cover_urls(self, metadata):
+        """Build cover ulrs for the related document."""
+        url_builder = current_app.config.get(
+            "ILS_LITERATURE_COVER_URLS_BUILDER")
+
+        # NOTE: we need the actual document here, bacause covers depend on,
+        # items or indentifiers.
+
+        Document = current_app_ils.document_record_cls
```

Comment:
This cannot be done like this because you are performing a db query for each search result: this will drastically slow down searches. You can use only the available metadata.
You can add the `identifiers` in the json resolver, as commented above, and use that.